### DISCUSSION TITLE
Illegal instruction - Raspberry PI 4 - aarch64

### DISCUSSION THREAD

**Discussion by qMalte** [openai/whisper#476] [Q&A] [answered]
Hello together,

I have installed Whisper on my Raspberry PI 4 (aarch64), but I get the following error when trying to translate a file to text:
`/usr/local/lib/python3.9/dist-packages/whisper/transcribe.py:78: UserWarning: FP16 is not supported on CPU; using FP32 instead
  warnings.warn("FP16 is not supported on CPU; using FP32 instead")
Illegal instruction`

I would be very happy to hear your ideas and suggestions for solutions.

Many greetings
Malte

**Comment by NeroKnox** [comment]
I've had the same problem with another aarch64 device. While I've gotten rid of the `UserWarning` by adding `--fp16 False` to the command, I still get the `Illegal instruction` error.
I would greatly appreciate help with this issue as well.

**Comment by longglecc** [comment]
I've had the same problem with another aarch64 device.
![image](https://user-images.githubusercontent.com/33145006/207216640-2a7a6ae2-9305-4599-aebc-338bd5555e40.png)

**Comment by qMalte** [comment, marked as answer]
Hello all,

the problem in my case will most likely be that the Raspberry PI does not have enough power to run Whisper correctly. The error message Illegal Instruction is to be understood as a general system crash.
I have installed Whisper on another arm64 device, which is much more powerful in both CPU and RAM, and it works fine. 

As a workaround I set up a vServer with enough resources and wrote a small Python script which receives an audio file via WebSockets, runs Whisper over it and returns a text.

If there is interest for such a script, I can publish it.


Many greetings 
Malte

**Reply by overcyber** [reply]
Hello, I would like the script please. I'm having the same problem with khadas vim3 pro, even with 6 cores whisper doesn't work in python only in C

**Comment by longglecc** [comment]
pkgs/main does not support Raspberry Pi, but conda-forge supports Raspberry Pi

**Comment by gitmanbloggus** [comment]
the raspberry pi is powerful enough to run a ai model with a overclock, so can't whisper just be ported?
